### ISSUE TITLE
addAsyncValidator: beforeSend function to hande ajax data option and other who need element state

### ISSUE THREAD

**Issue by robgha01** [guillaumepotier/Parsley.js#735]
Fix forked: https://github.com/guillaumepotier/Parsley.js/issues/736

I have the following code:
```addAsyncValidator('validzipcode',
    function (value) {
        console.log(this.$element); // jQuery Object[ input[name="q"] ]

        return 404 === value.status;
    }, "/ServiceMethods.asmx/SsnIsValid", { type: "POST", data: JSON.stringify({ ssn: ??? }), dataType: "json", contentType: "application/json; charset=utf-8" });```

I cant figure out how to pass the value from the input that i have decoreted the validator with to the data -> ssn object i first tried to write "value" and then parsley would change that to the input value but it dident work that way how is it possible to pass the value to ssn without calling $("element").val() ?

To solve this maybe a new attribute can be added where we get the opportunity to set the name of the value that will get sent to the server or to use some sort of magic word that get replaced ex: "parsleyValue" something like that ?

**Comment by marcandre** [comment]
Related to #627

**Comment by marcandre** [comment]
Now fixed, see #894
